experimental/QuadFormAndIsom/test/runtests.jl

```diff
@@ -0,0 +1,261 @@
+  V = ambient_space(D5)
+  Vf = @inferred quadratic_space_with_isometry(V, neg=true)
+  @test is_one(-isometry(Vf))
+  @test order_of_isometry(Vf) == 2
+  @test space(Vf) === V
+
+  for func in [rank, dim, gram_matrix, det, discriminant, is_positive_definite,
+              is_negative_definite, is_definite, diagonal, signature_tuple]
+    k = func(Vf)
+    @test k == func(V)
+  end
+
+  @test evaluate(minimal_polynomial(Vf), -1) == 0
+  @test evaluate(characteristic_polynomial(Vf), 0) == 1
+
+  G = matrix(FlintQQ, 6, 6 ,[3, 1, -1, 1, 0, 0, 1, 3, 1, 1, 1, 1, -1, 1, 3, 0, 0, 1, 1, 1, 0, 4, 2, 2, 0, 1, 0, 2, 4, 2, 0, 1, 1, 2, 2, 4])
+  V = quadratic_space(QQ, G)
+  f = matrix(QQ, 6, 6, [1 0 0 0 0 0; 0 0 -1 0 0 0; -1 1 -1 0 0 0; 0 0 0 1 0 -1; 0 0 0 0 0 -1; 0 0 0 0 1 -1])
+  Vf = @inferred quadratic_space_with_isometry(V, f)
+  @test order_of_isometry(Vf) == 3
+  @test order_of_isometry(rescale(Vf, -3)) == 3
+  L = lattice(rescale(Vf, -2))
+  @test is_even(L)
+  @test is_negative_definite(L) == is_positive_definite(Vf)
+
+  @test rank(biproduct(Vf, Vf)[1]) == 12
+  @test order_of_isometry(direct_sum(Vf, Vf, Vf)[1]) == 3
+  @test det(direct_product(Vf, Vf)[1]) == det(Vf)^2
+
+  @test Vf != quadratic_space_with_isometry(V, neg=true)
+  @test length(unique([Vf, quadratic_space_with_isometry(V, isometry(Vf))])) == 1
+  @test Vf^(order_of_isometry(Vf)+1) == Vf
+
+  V = quadratic_space(QQ, matrix(QQ, 0, 0, []))
+  Vf = @inferred quadratic_space_with_isometry(V)
+  @test order_of_isometry(Vf) == -1
+end
+
+@testset "Lattices with isometry" begin
+  A3 = root_lattice(:A, 3)
+  agg = automorphism_group_generators(A3, ambient_representation = false)
+  agg_ambient = automorphism_group_generators(A3, ambient_representation = true)
+  f = rand(agg)
+  g_ambient = rand(agg_ambient)
+
+  L = integer_lattice(gram = matrix(QQ, 0, 0, []))
+  Lf = integer_lattice_with_isometry(L, neg = true)
+  @test order_of_isometry(Lf) == -1
+
+  L = @inferred integer_lattice_with_isometry(A3)
+  @test length(unique([L, L, L])) == 1
+  @test ambient_space(L) isa QuadSpaceWithIsom
+  @test isone(isometry(L))
+  @test isone(ambient_isometry(L))
+  @test isone(order_of_isometry(L))
+  @test order(image_centralizer_in_Oq(L)) == 2
+
+  for func in [rank, genus, basis_matrix, is_positive_definite,
+               gram_matrix, det, scale, norm, is_integral, is_negative_definite,
+               degree, is_even, discriminant, signature_tuple, is_definite]
+    k = @inferred func(L)
+    @test k == func(A3)
+  end
+
+  LfQ = @inferred rational_span(L)
+  @test LfQ isa QuadSpaceWithIsom
+  @test evaluate(minimal_polynomial(L), 1) == 0
+  @test evaluate(characteristic_polynomial(L), 0) == -1
+
+  @test minimum(L) == 2
+  @test is_positive_definite(L)
+  @test is_definite(L)
+
+  nf = multiplicative_order(f)
+  @test_throws ArgumentError integer_lattice_with_isometry(A3, zero_matrix(QQ, 0, 0))
+
+  L2 = @inferred integer_lattice_with_isometry(A3, f, ambient_representation = false)
```

Comment:
```suggestion
  L2 = @inferred integer_lattice_with_isometry(A3, f; ambient_representation = false)
```
semicolon before kwargs (here and at other places) helps readability